app/iot/service.py

```diff
@@ -26,24 +26,24 @@ class IOTService:
     def __init__(self) -> None:
         self.devices: dict[str, Device] = {}
 
-    def register_device(self, device: Device) -> str:
-        device.connect()
+    async def register_device(self, device: Device) -> str:
+        await device.connect()
         device_id = generate_id()
         self.devices[device_id] = device
         return device_id
 
-    def unregister_device(self, device_id: str) -> None:
-        self.devices[device_id].disconnect()
+    async def unregister_device(self, device_id: str) -> None:
```

Comment:
The `unregister_device` method correctly uses `await` to asynchronously disconnect the device. Ensure that the device classes implement the `disconnect` method as an asynchronous function, which is already done in the `devices.py` file.